### PR TITLE
feat: add rust binding for `DuplexChallenger`

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This adds rust binding for `DuplexChallenger`

### CODE REVIEW

File: tachyon/c/zk/air/plonky3/challenger/baby_bear_poseidon2_duplex_challenger.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+TACHYON_C_EXPORT tachyon_plonky3_baby_bear_poseidon2_duplex_challenger*
+tachyon_plonky3_baby_bear_poseidon2_duplex_challenger_create();
+
+/**
+ * @brief Clones an existing duplex challenger structure.
+ *
+ * Creates a deep copy of the given duplex challenger structure.
+ *
+ * @param evals A pointer to the duplex challenger structure to clone.
```
Comment (ashjeong) [inline tachyon/c/zk/air/plonky3/challenger/baby_bear_poseidon2_duplex_challenger.h:32]:
```suggestion
 * @param challenger A pointer to the duplex challenger structure to clone.

File: tachyon/c/zk/air/plonky3/challenger/baby_bear_poseidon2_duplex_challenger.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @return A pointer to the cloned duplex challenger structure.
+ */
+TACHYON_C_EXPORT tachyon_plonky3_baby_bear_poseidon2_duplex_challenger*
+tachyon_plonky3_baby_bear_poseidon2_duplex_challenger_clone(
+    tachyon_plonky3_baby_bear_poseidon2_duplex_challenger* challenger);
+
+/**
+ * @brief Destroys a duplex challenger, freeing its resources.
+ *
+ * @param challenger The duplex challenger to destroy.
```
Comment (ashjeong) [inline tachyon/c/zk/air/plonky3/challenger/baby_bear_poseidon2_duplex_challenger.h:42]:
```suggestion
 * @param challenger A pointer to the duplex challenger to destroy.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

File: tachyon/c/zk/air/plonky3/challenger/baby_bear_poseidon2_duplex_challenger_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  void TearDown() override {
+    tachyon_plonky3_baby_bear_poseidon2_duplex_challenger_destroy(challenger_);
+  }
+
+ protected:
+  tachyon_plonky3_baby_bear_poseidon2_duplex_challenger* challenger_ = nullptr;
+};
+
+TEST_F(DuplexChallengerTest, APIs) {
+  zk::air::plonky3::DuplexChallenger<Poseidon2, 16, 4> challenger =
```
Comment (ashjeong) [inline tachyon/c/zk/air/plonky3/challenger/baby_bear_poseidon2_duplex_challenger_unittest.cc:31]:
```suggestion
DuplexChallenger<Poseidon2, 16, 4> challenger =

**Review by ashjeong — CHANGES_REQUESTED** [review]
Instead of "for c API" in the commit titles, could you change it to "to c API"?

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**